plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt

```diff
@@ -663,16 +664,29 @@ abstract class AnnotationManagerImpl<G : Geometry, T : Annotation<G>, S : Annota
       return false
     }
 
+    private val onMapIdleListener: OnMapIdleListener = object : OnMapIdleListener {
+      override fun onMapIdle() {
+        mapListenerDelegate.removeOnMapIdleListener(this)
+        // Remove dragging annotation from drag layer
+        Handler(Looper.getMainLooper()).postDelayed(
```

Comment:
could create handler only once and store it as a val

Reply:
also what if user starts dragging again while we're waiting for one second here?
shouldn't we do `handler.removeCallbackAndMessages(null)` when drag starts?

Reply:
If users drag again, `draggingAnnotation` will not be null, that's simply what we do while dragging. You can see from the attached video, the time between two drags is obviously less than 1 second and it works well.

Reply:
The problem that it does not 😄 
https://github.com/mapbox/mapbox-maps-android/pull/639#issuecomment-918855680